lib/resque.rb

```diff
@@ -147,6 +147,15 @@ def redis_id
     data_store.identifier
   end
 
+  def stat_data_store=(stat_data_store)
+    stat = Resque::Stat.data_store = stat_data_store
+    @stat_data_store = stat
+  end
+
+  def stat_data_store
+    @stat_data_store ||= Resque::Stat
```

Comment:
The return of this method should be `Resque::Stat.data_store` otherwise you will get either an object that respond to those 4 methods we know or a object that respond to all methods `Resque::Stat` respond to. It will be inconsistent.